packages/clarity-js/src/layout/mutation.ts

```diff
@@ -169,19 +170,22 @@ function track(m: MutationRecord, timer: Timer): string {
     // In those cases, IDs will change however the selector (which is relative to DOM xPath) remains the same
     let key = [parent, element, m.attributeName, names(m.addedNodes), names(m.removedNodes)].join();
     // Initialize an entry if it doesn't already exist
-    history[key] = key in history ? history[key] : [0];
+    history[key] = key in history ? history[key] : [0, instance];
     let h = history[key];
     // Lookup any pending nodes queued up for removal, and process them now if we suspended a mutation before
-    if (inactive === false && h[0] >= Setting.MutationSuspendThreshold) { processNodeList(h[1], Source.ChildListRemove, timer); }
+    if (inactive === false && h[0] >= Setting.MutationSuspendThreshold) { processNodeList(h[2], Source.ChildListRemove, timer); }
     // Update the counter
-    h[0] = inactive ? h[0] + 1 : 1;
+    h[0] = inactive ? (h[1] === instance ? h[0] : h[0] + 1) : 1;
```

Comment:
Style Suggestion: Add a brief comment to indicate what h[0/1/2] represent. Maybe declare constants for the indices to indicate what each represents.

Reply:
It's defined for the type "MutationHistory" with comments in layout.d.ts.